plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -862,20 +872,50 @@ class FossId internal constructor(
                     }
                 }
 
-                emit(file to filteredSnippets.toSet())
+                val fileWithoutPrefix = archivePrefix?.let { file.removePrefix(it) } ?: file
+
+                emit(fileWithoutPrefix to filteredSnippets.toSet())
             }
         }
 
-        return RawResults(
+        return RawResults.createAndRemovePrefix(
             identifiedFiles,
             markedAsIdentifiedFiles,
             listIgnoredFiles,
             pendingFiles,
             snippets,
-            matchedLines
+            matchedLines,
+            archivePrefix
         )
     }
 
+    /**
+     * When the scan is created in Archive upload mode, all the file paths are prefixed with the archive name e.g.
+     * fossid-source-archive832364312005325574.zip/test.txt. This function searches this prefix in the list of pending
+     * files or in the list of identified files. If no such prefix is found, null is returned.
+     */
+    internal fun getArchivePrefix(
+        pendingFiles: List<String>,
+        identifiedFiles: List<IdentifiedFile>,
+        markedAsIdentifiedFiles: List<MarkedAsIdentifiedFile>,
+        listIgnoredFiles: PolymorphicList<IgnoredFile>
+    ): String? {
+        val prefix = pendingFiles.findPrefix { it }
+            ?: identifiedFiles.findPrefix { it.file.path }
+            ?: markedAsIdentifiedFiles.findPrefix { it.file.path }
+            ?: listIgnoredFiles.findPrefix { it.path }
+
+        return prefix?.let {
+            logger.info { "Found archive prefix '$prefix' from snippets." }
+            "$prefix/"
+        }
+    }
+
+    fun <T> List<T>.findPrefix(selector: (T) -> String?): String? =
+        firstOrNull()
+            ?.takeIf { selector(it)?.startsWith("fossid-source-archive") == true }
+            ?.let { selector(it)?.substringBefore("/") }
```

Comment:
In my example in made this an inner function of `getArchivePrefix` on purpose, because it's not needed anywhere else. At least the function should be private.

Reply:
Yes class context is not needed. So is an inner function better ?

Reply:
Actually it should then be a top-level function, as an inner function also would have class context. But I do not insist on the change, and maybe we should just keep it as-is to move things forward.